google-cloud-spanner/src/main/java/com/google/cloud/spanner/TransactionRunnerImpl.java

```diff
@@ -286,6 +300,10 @@ void commit() {
     volatile ApiFuture<CommitResponse> commitFuture;
 
     ApiFuture<CommitResponse> commitAsync() {
+      checkState(
```

Comment:
Ah, good point. I didn't see any failure in "mvn test" but will modify the code accordingly.

Reply:
My bad, the `inlineCommitState` will obviously NOT be `SUCCEEDED` here if `AsyncTransactionManager` is being used, so this check will not fail. But (while looking into this):
1. Another problem with the current `InlineCommit` option is that it is defined as a `TransactionOption`. Those can also be specified when creating a `TransactionManager` or `AsyncTransactionManager`. What happens when a user does that? As far as I can tell from the code, it will be ignored, but that might also be a little confusing for users.

Reply:
changed the error message to be more generic.
Removed Options.inlineCommitOption() so users should not be able to set it.